src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -222,6 +222,21 @@
   [max-frame-size url]
   (format "%s&wireFormat.maxFrameSize=%s&marshal=true" url max-frame-size))
 
+(defn- transfer-old-messages! [connection]
+  (let [[pending exists?]
```

Comment:
Looks like we might be able to use [this](http://activemq.apache.org/maven/5.10.0/apidocs/org/apache/activemq/broker/jmx/BrokerViewMBean.html#getQueues()) JMX call to get a list of queues that the broker knows about. Then we wouldn't need to infer existence based on queue size (and handle the errors associated with that etc).